lucene/src/main/java/org/hibernate/hql/lucene/internal/ClassBasedLuceneQueryResolverDelegate.java

```diff
@@ -313,7 +313,7 @@ public void propertyPathCompleted(PropertyPath path) {
 
 	@Override
 	public void registerJoinAlias(Tree alias, PropertyPath path) {
-		if ( !path.getNodes().isEmpty() && !aliasToPropertyPath.containsKey( alias ) ) {
+		if ( !path.getNodes().isEmpty() && !aliasToPropertyPath.containsKey( alias.getText() ) ) {
```

Comment:
sure, but we normally call such mis-coordinations "bug" not typo, so a JIRA would be nice ;-)
What about scheduling, do you need this backported somewere and/or released soon?